### PR TITLE
Switch to Typescript

### PR DESCRIPTION
**Pull request by charlag**

### CODE REVIEW

File: lib/attachment.js [hunk 1]
```diff
@@ -1,92 +0,0 @@
...
-	) {
-		this.data = data
-		this.fileName = fileName
-		this.type = type
-		this.renderingPosition = renderingPosition
-		this.contentId = contentId
-		this.isContactPhoto = isContactPhoto
-	}
-
-	writeProperties(storage: any, index: number): number {
```
Comment (ganthern) [inline lib/attachment.js:35]: explicit any

Comment (charlag) [reply lib/attachment.js:35]: this is deleted?

Comment (ganthern) [reply lib/attachment.js:35]: yes, I mixed up the files

File: lib/helpers/strings.ts [hunk 2]
```diff
@@ -1,35 +1,39 @@
...
-            } else { // unicode. may consist of multiple code points
-                for(const codepoint of glyph.split('')) {
+            } else {
+                // unicode. may consist of multiple code points
+                for (const codepoint of glyph.split("")) {
                     // TODO:
                     // RTF control words generally accept signed 16-bit numbers as arguments.
                     // For this reason, Unicode values greater than 32767 must be expressed as negative numbers.
                     rtfEscaped.push("\\u")
+                    // @ts-ignore: something weird is happening here
```
Comment (ganthern) [inline lib/helpers/strings.ts]: I'm getting 
`[!] (plugin typescript) Error: @rollup/plugin-typescript TS2345: Argument of type 'number' is not assignable to parameter of type 'string'.
`
without the ignore. that error seems reasonable?

File: lib/index.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export {utf16LeArrayToString, utf8ArrayToString, bigInt64FromParts} from "./utils/utils.js"
```
Comment (ganthern) [inline lib/index.ts:1]: is it correct to import from .js files here?

Comment (charlag) [reply lib/index.ts:1]: yes, js spec says that the only valid way to import files (as opposed to modules) and TS won't let you import .ts files

File: lib/helpers/strings.ts (hunk 2 above)
Comment (charlag) [reply lib/helpers/strings.ts]: Yeah I just wanted to ask you about what should happen there, looks like a bug

Comment (ganthern) [reply lib/helpers/strings.ts]: we want to escape unicode codepoints `"🍆"` -> `"\\u55356?\\u57158?"` as specced in the [rtf 1.5 spec](https://www.biblioscape.com/rtf15_spec.htm#Heading9).

the `?` is the "equivalent character(s) in ANSI representation" mentioned for the `\uN` control word, so non-unicode readers will show a `?`

File: lib/utils/mime.ts [hunk 4]
```diff
@@ -0,0 +1,567 @@
+const MimeTypes = Object.freeze({
```
Comment (ganthern) [inline lib/utils/mime.ts:1]: should this also be converted to an enum?

**Review by ganthern — APPROVED** [review]

**Pull request merged**